rasa/core/policies/flow_policy.py

```diff
@@ -178,20 +194,33 @@ def predict_action_probabilities(
              The prediction.
         """
         if not self.supports_current_stack_frame(tracker):
+            # if the policy doesn't support the current stack frame, we'll abstain
             return self._prediction(self._default_predictions(domain))
 
         flows = flows or FlowsList([])
         executor = FlowExecutor.from_tracker(tracker, flows, domain)
 
         # create executor and predict next action
-        prediction = executor.advance_flows(tracker)
-        return self._create_prediction_result(
-            prediction.action_name,
-            domain,
-            prediction.score,
-            prediction.events,
-            prediction.metadata,
-        )
+        try:
+            prediction = executor.advance_flows(tracker)
+            return self._create_prediction_result(
+                prediction.action_name,
+                domain,
+                prediction.score,
+                prediction.events,
+                prediction.metadata,
+            )
+        except FlowCircuitBreakerTrippedException as e:
+            structlogger.error(
+                "flow.circuit_breaker",
+                dialogue_stack=e.dialogue_stack,
+                number_of_steps_taken=e.number_of_steps_taken,
+                event_info=(
+                    "The flow circuit breaker tripped. "
+                    "There appears to be an infinite loop in the flows."
+                ),
+            )
+            return self._prediction(self._default_predictions(domain))
```

Comment:
that just means remain silent, right? Seems like this might warrant a message to the user as well - e.g. with the internal error pattern. Could also be tested then.

Reply:
mhm yeah that is a good point. I followed the approach we took with the circuit breaker for actions but I agree that we should also inform the user.

I'll add an error stack frame :+1: